packages/manager/apps/vrack-services/src/components/OrderSubmitModalContent.component.tsx

```diff
@@ -53,38 +49,23 @@ export const OrderSubmitModalContent: React.FC<OrderSubmitModalContentProps> = (
   const { t } = useTranslation('vrack-services');
   const { trackClick } = useOvhTracking();
   const [isContractAccepted, setIsContractAccepted] = React.useState(false);
-  const { mutate: sendOrder, isPending, error, isError } = useMutation<
-    ApiResponse<Order>,
-    ApiError
-  >({
-    mutationFn: () =>
-      postOrderCartCartIdCheckout({
-        cartId,
-        autoPayWithPreferredPaymentMethod: true,
-        waiveRetractationPeriod: true,
-      }),
-    onSuccess,
-    onError: async (response) => {
-      const {
-        request: { status },
-      } = response;
+  const {
+    sendOrder,
+    isPending,
+    error,
+    isError,
+    data,
+    sendOrderState,
+  } = useSendOrder();
```

Comment:
We could take the onSucccess and onError callbacks and pass them to the useMutation inside the hook maybe